conf/modules.config

```diff
@@ -17,173 +17,91 @@ process {
-            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+    /* 
+    ----------
+    ASSEMBLY
+    ----------
+    */
+    // FLYE: Here I can switch to nf-core/flye
+    withName: FLYE {
+        ext.ags = [
+            "--genome-size ${params.genome_size}",
+            params.flye_args
         ]
-    }
-    withName: 'GENESCOPEFK' {
-        ext.args   = { [
-            "--kmer_length ${meta.kmer_size?: params.kmer_size}",
-            "--ploidy ${meta.ploidy?: params.ploidy}"
-        ].join(' ') }
-        publishDir = [
-            path: { "$params.outdir/genome_properties/genescopefk" },
-            mode: params.publish_dir_mode,
-            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        .join(" ")
+        .trim(  )
+    publishDir = [
+        path: { "${params.outdir}/assembly/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
         ]
     }
-    withName: 'MERQURYFK_PLOIDYPLOT' {
+    // HIFIASM:
+    /* Don't think I should switch to nf-core 
+       here since the UL process takes more inputs */
+    withName: 'HIFIASM.*' {
+        ext.args = [ params.hifiasm_args ]
         publishDir = [
-            path: { "$params.outdir/genome_properties/ploidyplot" },
+            path: { "${params.outdir}/assembly/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
             mode: params.publish_dir_mode,
             saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
         ]
     }
-    withName: 'MERQURYFK_KATGC' {
+    /* 
+    ----------
+    QC
+    ----------
+    */
+    // BUSCO: switch to nf-core
+    withName: BUSCO {
+        // Subworkflow: "${task.process.tokenize(':')[-3].toLowerCase()}"
+        ext.prefix = '"${meta.id}"-"${task.process}.tokenize(":")[-3].toLowerCase()"-"${lineage}"'
         publishDir = [
-            path: { "$params.outdir/genome_properties/katgc" },
+            path: { "${params.outdir}/QC/${task.process.tokenize(':')[-3].toLowerCase()}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
             mode: params.publish_dir_mode,
-            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }            
         ]
     }
-
-    // COMPARE_LIBRARIES
-    // withName: 'MERQURYFK_KATCOMP' {
-    //     publishDir = [
-    //         path: { "$params.outdir/library_comparison/kat_comp" },
-    //         mode: params.publish_dir_mode,
-    //         saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
-    //     ]
-    // }
-
-    // CONTAMINATION SCREEN
-    withName: 'MASH_SCREEN' {
-        ext.args   = '-w'
-        ext.prefix = { "${meta.id}_${query.baseName}" }
```

Comment:
These outer curly brackets here have a specific function: it executes the code only at runtime and by that making sure that params are parsed at runtime and not during config reading. That is optimal. You seem to not use that. This trick is missing in many configs of other pipelines (historic reasons), but if you do it new, why not optimal ;)

Reply:
Ah, great, thanks for the explanation, I was wondering how the behaviour is controlled while writing the config and arrived at using `''` to try to control when things are substituted, I will use `{}` (see also your comment below)